### PR TITLE
Add ability to configure the encryption_key_loc filesystem

### PR DESCRIPTION
**Pull request by kadaan**
Fix for #731

### CODE REVIEW

**Comment by kadaan** [comment]
# New Configuration
`encrypt.key.fs.uri` is optional.  If it is not provided, then the code will perform as it did before.

# Example Configuration
To allow encrypted password to be used when gobblin in run under yarn, you need to make sure that the master password file is copied to the app_master and containers.  This is done by adding `/opt/apps/gobblin/current/master.pass` to `gobblin.yarn.app.master.files.local` and `gobblin.yarn.container.files.local`.  Additionally, `encrypt.key.fs.uri` needs to specify the local filesystem `"file:///"` and `encrypt.key.loc` need specify the master password file at the root of the current directory `./master.pass`.

See the example below:
```
gobblin.yarn.app.master.files.local=${gobblin.yarn.conf.dir}"/log4j-yarn.properties,"${gobblin.yarn.conf.dir}"/application.conf,"${gobblin.yarn.conf.dir}"/reference.conf,{{ app_current_dir }}/master.pass"
gobblin.yarn.container.files.local=${gobblin.yarn.app.master.files.local}
...
encrypt.key.fs.uri="file:///"
encrypt.key.loc=./master.pass
```

**Comment by kadaan** [comment]
@sahilTakiar @zliu41 Can you please review this?

File: gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java [hunk 1]
```diff
@@ -501,6 +501,7 @@
   /**
    * Password encryption and decryption properties.
    */
+  public static final String ENCRYPT_KEY_FS_URI = "encrypt.key.fs.uri";
```
Comment (chavdar) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: I am not sure we need a new configuration property. ENCRYPT_KEY_LOC can just contain a qualified path.

**Comment by kadaan** [comment]
### The master.pass file exists in the current directory
screen shot 2016-02-25 at 1 01 43 pm

### But I can't get it like this
screen shot 2016-02-25 at 1 02 00 pm

### Or like this
screen shot 2016-02-25 at 1 02 31 pm

### But this works
screen shot 2016-02-25 at 1 03 10 pm

File: gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java (hunk 1 above)
Comment (kadaan) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: @chavdar See the comment below.

**Comment by chavdar** [comment]
OK. Digging through the Hadoop source code I found:

    /**
     * Pathnames with scheme and relative path are illegal.
     * @param path to be checked
     */
    void checkNotSchemeWithRelative() {

LGTM.

**Pull request merged**